### ISSUE TITLE
`i18n` function is not referentially stable

### ISSUE THREAD

**Issue by peeke** [kaliberjs/i18n#2]
`useI18n()` returns a different function each time it's called. This makes it hard to use the i18n function inside a `useEffect`.

**Comment by EECOLOR** [comment]
Do you have an example of an effect where you would want to use `i18n`?

**Comment by peeke** [comment]
I had a `useEffect`, where I wanted to use `i18n` inside. Had to pass the `i18n` as a dependency, or store it in a ref (which felt a bit hacky). Ended up taking a different route altogether, but it felt like something that should maybe be possible.

**Comment by EECOLOR** [comment]
I agree it should be possible. While thinking about it I wondered of a use case where you would need `i18n` in an effect. Rendering usually happens in the 'render' function, so it might indicate unneeded complexity.

**Comment by peeke** [comment]
That is what actually was the case, and I fixed it by doing exactly that. We could also just note the fact that it isn't referentially stable in the docs.

**Comment by EECOLOR** [comment]
I actually think it should be referentially more stable. So I would vote to add a `useCallback` around it.

That said, we should probably use it as a potential red flag when we see it used inside another hook during reviews.
